app/services/data_migrations/remove_duplicate_provider.rb

```diff
@@ -0,0 +1,30 @@
+module DataMigrations
+  class RemoveDuplicateProvider
+    TIMESTAMP = 20210505095417
+    MANUAL_RUN = false
+
+    def change
+      permissions = ProviderRelationshipPermissions
+        .joins('INNER JOIN providers tp ON training_provider_id = tp.id')
+        .joins('INNER JOIN providers rp ON ratifying_provider_id = rp.id')
+        .where('tp.name = rp.name')
+
+      if permissions.any?
+        permissions.each do |permission|
+          ActiveRecord::Base.transaction do
+            courses = permission.training_provider.courses.where(accredited_provider: permission.ratifying_provider)
+
+            next if permission.ratifying_provider.courses.any?
```

Comment:
This check can move to the top of the transaction.

Do we also need to check that the provider we are about to delete is not part of other relationships and whether it has any `accredited_courses`? We should check we aren't going to orphan any provider users too.